catalog/MDCCatalog/MDCCatalogThemer.m

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+#import <Foundation/Foundation.h>
+
+#import "MDCCatalogThemer.h"
+#import "MDCActivityIndicatorColorThemer.h"
+#import "MDCAlertColorThemer.h"
+#import "MDCButtonBarColorThemer.h"
+#import "MDCButtonColorThemer.h"
+#import "MDCFeatureHighlightColorThemer.h"
+#import "MDCFlexibleHeaderColorThemer.h"
+#import "MDCHeaderStackViewColorThemer.h"
+#import "MDCNavigationBarColorThemer.h"
+#import "MDCPageControlColorThemer.h"
+#import "MDCProgressViewColorThemer.h"
+#import "MDCSliderColorThemer.h"
+#import "MDCTabBarColorThemer.h"
+#import "MDCTextFieldColorThemer.h"
+#import "MaterialTextFields.h"
+
+@implementation MDCCatalogThemer
+
++ (void)applyColorScheme:(NSObject<MDCColorScheme> *)colorScheme {
+  // Apply color scheme to material design components using component themers.
+  [MDCActivityIndicatorColorThemer applyColorScheme:colorScheme
+                                toActivityIndicator:MDCActivityIndicator.appearance];
+  [MDCAlertColorThemer applyColorScheme:colorScheme];
+  [MDCButtonBarColorThemer applyColorScheme:colorScheme toButtonBar:MDCButtonBar.appearance];
+  // When a button bar is contained in a navigation bar do not color the button bar. This is useful
+  // for when an app bar with imagery in the flexible header has a button bar. If the button bar has
+  // a color then the imagery is obstucted by the solid color of the button bar.
+#if defined(__IPHONE_9_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
+  NSOperatingSystemVersion iOS9Version = {9, 0, 0};
+  if ([NSProcessInfo.processInfo isOperatingSystemAtLeastVersion:iOS9Version]) {
+    MDCButtonBar *appearanceProxy =
+        [MDCButtonBar appearanceWhenContainedInInstancesOfClasses:@[MDCNavigationBar.class]];
+    appearanceProxy.backgroundColor = nil;
+  } else {
+#endif
+    [MDCButtonBar appearanceWhenContainedIn:MDCNavigationBar.class, nil].backgroundColor = nil;
+#if defined(__IPHONE_9_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_9_0)
+  }
+#endif
```

Comment:
Do you still need this code? since you are setting the background color to nil?

Reply:
Nope, good catch. Thanks!